### PR TITLE
Issue/4525 reader sort by tagged date

### LINKED ISSUE #4525: Reader: sort tag streams by tagged date
Based on [this comment](https://github.com/Automattic/wp-calypso/issues/7731#issuecomment-244070453) from @blowery, tag streams in the reader should be sorted by the date they were tagged because this is the order used by the backend.

### PR DESCRIPTION
**Pull request by nbradbury**
Fixes #4525 - based on [this comment](https://github.com/Automattic/wp-calypso/issues/7731#issuecomment-244070453) and on Slack discussions, posts in reader tag streams should be sorted by the date they were tagged. This is how they're sorted on the backend, and not doing this in the app (we sort by pubDate) results in some posts appearing out of order compared to Calypso.

This PR adds a generic `date` field to the post model which is set to the tagged date when available. This date field is then used to determine how posts are sorted, and is used when requesting older posts (the `before` param passed to `/read/tags/{tagName}/posts`).

The `published` date, which existed previously, is now only used for display.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java [hunk 1]
```diff
@@ -39,8 +40,9 @@
     private String primaryTag;    // most popular tag on this post based on usage in blog
     private String secondaryTag;  // second most popular tag on this post based on usage in blog
 
-    public double sortIndex;
-    private String published;
+    public double sortIndex;      // determines how posts are sorted for display
+    private String date;          // generic date - depends on the stream the post is from
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java]: Can you be more specific in this comment? Add an example use case (like `which is set to the tagged date when available`)?

Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java]: Or just add "used for sorting posts".

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java]: @maxme I updated the comment in ed312b1.

**Comment by maxme** [comment]
:shipit:

**Pull request merged**